src/tls/keypair.h

```diff
@@ -61,37 +61,17 @@ namespace tls
   using Hash = std::array<uint8_t, MD_SIZE>;
 
   inline bool verify_secp256k_bc(
-    secp256k1_context* ctx, const uint8_t* signature, const uint8_t* hash)
+    secp256k1_context* ctx,
+    const uint8_t* signature,
+    const uint8_t* hash,
+    const secp256k1_pubkey* public_key)
   {
-    secp256k1_pubkey public_key;
     secp256k1_ecdsa_recoverable_signature sig;
-
-    if (
-      secp256k1_ecdsa_recoverable_signature_parse_compact(
-        ctx, &sig, signature, signature[REC_ID_IDX]) != 1)
-    {
-      LOG_INFO_FMT(
-        "secp256k1_ecdsa_recoverable_signature_parse_compact failed");
-      return false;
-    }
-
+    int rc = secp256k1_ecdsa_recoverable_signature_parse_compact(
```

Comment:
We don't use or check rc here. Should we still be failing if `parse_compact` or `signature_convert` fail?

Reply:
This is the old code, and it never handled these errors, but it definitely should. Good catch.